ddtrace/tracer/textmap.go

```diff
@@ -487,3 +501,292 @@ func (*propagatorB3) extractTextMap(reader TextMapReader) (ddtrace.SpanContext,
+	}
+	return &ctx, nil
+}
+
+// parseTraceparent attempts to parse traceparentHeader which describes the position
+// of the incoming request in its trace graph in a portable, fixed-length format.
+// The format of the traceparentHeader is `-` separated string with in the
+// following format: `version-traceId-spanID-flags`,
+// where:
+// - version - represents the version of the W3C Tracecontext Propagation format in hex format.
+// - traceId - represents the propagated traceID in the format of 32 hex-encoded digits.
+// - spanID - represents the propagated spanID in the format of 16 hex-encoded digits.
+// - flags - represents the propagated flags in the format of 2 hex-encoded digits, and supports 8 unique flags.
+// Example value of HTTP `traceparent` header: `00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01`,
+// Currently, Go tracer doesn't support 128-bit traceIDs, so the full traceID (32 hex-encoded digits) must be
+// stored into a field that is accessible from the span’s context. TraceId will be parsed from the least significant 16
+// hex-encoded digits into a 64-bit number.
+func parseTraceparent(ctx *spanContext, header string) error {
+	nonWordRegex := "\\W_"
+	header = strings.ToLower(strings.Trim(header, "\t -"))
+	if len(header) == 0 {
+		return ErrSpanContextNotFound
+	}
+	if len(header) != 55 {
+		return ErrSpanContextCorrupted
+	}
+	parts := strings.Split(header, "-")
+	if len(parts) != 4 {
+		return ErrSpanContextCorrupted
+	}
+	version := strings.Trim(parts[0], nonWordRegex)
+	if len(version) != 2 {
+		return ErrSpanContextCorrupted
+	}
+	if v, err := strconv.ParseUint(version, 16, 64); err != nil || v == 255 {
+		return ErrSpanContextCorrupted
+	}
+	// parsing traceID
+	fullTraceID := strings.Trim(parts[1], nonWordRegex)
+	if len(fullTraceID) != 32 {
+		return ErrSpanContextCorrupted
+	}
+	// checking that the entire TraceID is a valid hex string
+	if ok, err := regexp.MatchString("^[a-f0-9]+$", fullTraceID); !ok || err != nil {
+		return ErrSpanContextCorrupted
+	}
+	var err error
+	if ctx.traceID, err = strconv.ParseUint(fullTraceID[16:], 16, 64); err != nil {
+		return ErrSpanContextCorrupted
+	}
+	if ctx.traceID == 0 {
+		if strings.Trim(fullTraceID[:16], "0") == "" {
+			return ErrSpanContextNotFound
+		}
+	}
+	// parsing spanID
+	spanID := strings.Trim(parts[2], nonWordRegex)
+	if len(spanID) != 16 {
+		return ErrSpanContextCorrupted
+	}
+	if ok, err := regexp.MatchString("[a-f0-9]+", spanID); !ok || err != nil {
+		return ErrSpanContextCorrupted
+	}
+	if ctx.spanID, err = strconv.ParseUint(spanID, 16, 64); err != nil || ctx.spanID == 0 {
+		return ErrSpanContextNotFound
+	}
+	// parsing flags
+	flags := parts[3]
+	f, err := strconv.ParseInt(flags, 16, 8)
+	if err != nil {
+		return ErrSpanContextCorrupted
+	}
+	// setting trace-id to be used for span context propagation
+	setPropagatingTag(ctx, w3cTraceIDTag, fullTraceID)
+	ctx.setSamplingPriority(int(f)&0x1, samplernames.Unknown)
```

Comment:
I see that only the 0x1 flag is specified currently, but is there a reason to explicitly mask other possible flags? 
We will have to update this if other flags are ever introduced.

Reply:
I could check if it's just `01` for now, but I think explicilty masking is more certain. And it will work in case some vendor sends a different value than `01` (in the case we haven't updated the code by that point)

Reply:
Nevermind my previous comment, I misread the code before. This is correct.